bin/download-with-mithril.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+set -x
+set -a
+
+# Create data directories if they don't exist
+mkdir -p "${SNAPSHOTS_CARDANO_NODE_DIR}" "${SNAPSHOTS_KUPO_DIR}" "${EXPORTED_SNAPSHOT_BASE_PATH_WITH_DATA_SOURCE}"
+
+MITHRIL_QUERY=$(mithril-client cardano-db snapshot list --json | jq -r '.[0]')
+# shellcheck disable=SC2034
+DIGEST=$(jq -r '.digest' <<< "${MITHRIL_QUERY}")
+MITHRIL_SNAPSHOT_NAME=$(jq -r '.beacon | "\(.network)-\(.epoch)-\(.immutable_file_number)"' <<< "${MITHRIL_QUERY}")
+# shellcheck disable=SC2034
+LOCAL_CARDANO_NODE_SNAPSHOT_DIR="${SNAPSHOTS_CARDANO_NODE_DIR}/${MITHRIL_SNAPSHOT_NAME}"
+
+# SNAPSHOT_DETAIL=$(mithril-client cardano-db snapshot show "${DIGEST}" --json)
+# CARDANO_NODE_VERSION=$(jq -r '.cardano_node_version' <<< "${SNAPSHOT_DETAIL}")
+
+# shellcheck disable=SC2034
+EXPORTED_KUPO_SNAPSHOT_PATH="${EXPORTED_SNAPSHOT_BASE_PATH_WITH_DATA_SOURCE}/${DATA_SOURCE}-${MITHRIL_SNAPSHOT_NAME}.tgz"
+# shellcheck disable=SC2034
+LOCAL_KUPO_DATA_PER_SNAPSHOT="${SNAPSHOTS_KUPO_DIR}/${MITHRIL_SNAPSHOT_NAME}"
+
+echo "Aggregator endpoint: ${AGGREGATOR_ENDPOINT}"
+echo "Genesis verification key: ${GENESIS_VERIFICATION_KEY}"
+
+echo "Deleting previous dir (if present): ${LOCAL_CARDANO_NODE_SNAPSHOT_DIR}"
+
+if [ ! -d "${LOCAL_CARDANO_NODE_SNAPSHOT_DIR}" ]; then
```

Comment:
So if the `mitrhil-client` finds the folder it is expecting to put data it just fails (without any check). 
In the previous approach, we were simply deleting the dir (that was what the previous/outdated  `echo` was suggesting). I've empirically found out that most of the time the download of the snapshot is ok (and even if it fails it will start again in max 2 days (it's not a big deal if we have like "a hole", is it?), for this reason I've removed the `rm -f` to have faster iterations(the idea is that we assume that if the folder is present the db in it is complete/intact).

The alternative (cheap) approach is moving back to deleting the folder that `mithril-client` will try to create: in that case we will never have invalid DB (but we might end up re-downloading valid DBs).

Implementing a more precise check on the DB I honestly think is overkill, what's your opinion on this?

Reply:
> Implementing a more precise check on the DB I honestly think is overkill, what's your opinion on this?

Agree, I would go for the cheap approach. You can use a bash' `trap` that removes the directory if `mithril-client cardano-db download <..>` exits with a non zero status.
If you agree please create an issue (or I will do it if we all agree) and ignore the problem in this PR.

Reply:
> even if it fails it will start again in max 2 days
Oh I was thinking it would fail indefinitely since the folder has been created and subsequent downloads will be skipped. If that's not the case, I'm fine with it as is